docs/database/includes/azure-postgresql-client.md

```diff
@@ -2,42 +2,167 @@
+
+<!-- TODO: Add xref to AddKeyedAzureNpgsqlDataSource when available -->
+
+There might be situations where you want to register multiple `NpgsqlDataSource` instances with different connection names. To register keyed Npgsql clients, call the `AddKeyedAzureNpgsqlDataSource` method:
+
+```csharp
+builder.AddKeyedAzureNpgsqlDataSource(name: "chat");
+builder.AddKeyedAzureNpgsqlDataSource(name: "queue");
+```
+
+Then you can retrieve the `NpgsqlDataSource` instances using dependency injection. For example, to retrieve the connection from an example service:
+
+```csharp
+public class ExampleService(
+    [FromKeyedServices("chat")] NpgsqlDataSource chatDataSource,
+    [FromKeyedServices("queue")] NpgsqlDataSource queueDataSource)
+{
+    // Use data sources...
+}
+```
+
+For more information on keyed services, see [.NET dependency injection: Keyed services](/dotnet/core/extensions/dependency-injection#keyed-services).
+
+#### Configuration
+
+The .NET Aspire Azure Npgsql integration provides multiple options to configure the database connection based on the requirements and conventions of your project.
+
+##### Use a connection string
+
+When using a connection string defined in the `ConnectionStrings` configuration section, you provide the name of the connection string when calling `AddAzureNpgsqlDataSource`:
 
 ```csharp
-builder.AddNpgsqlDataSource(
-    "postgresdb", 
-    configureDataSourceBuilder: (dataSourceBuilder) =>
+builder.AddAzureNpgsqlDataSource("postgresdb");
+```
+
+The connection string is retrieved from the `ConnectionStrings` configuration section, for example, consider the following JSON configuration:
+
+```json
 {
-    if (string.IsNullOrEmpty(dataSourceBuilder.ConnectionStringBuilder.Password))
-    {
-        var credentials = new DefaultAzureCredential();
-        var tokenRequest = new TokenRequestContext(["https://ossrdbms-aad.database.windows.net/.default"]);
-
-        dataSourceBuilder.UsePasswordProvider(
-            passwordProvider: _ => credentials.GetToken(tokenRequest).Token,
-            passwordProviderAsync: async (_, ct) => (await credentials.GetTokenAsync(tokenRequest, ct)).Token);
+  "ConnectionStrings": {
+    "postgresdb": "Host=myserver;Database=test"
+  }
+}
+```
+
+For more information on how to configure the connection string, see the [Npgsql connection string documentation](https://www.npgsql.org/doc/connection-string-parameters.html).
+
+> [!NOTE]
+> The username and password are automatically inferred from the credential provided in the settings.
+
+##### Use configuration providers
+
+<!-- TODO: Add xref to AzureNpgsqlSettings when available -->
+
+The .NET Aspire Azure Npgsql integration supports <xref:Microsoft.Extensions.Configuration>. It loads the `AzureNpgsqlSettings` from configuration using the `Aspire:Azure:Npgsql` key. For example, consider the following _appsettings.json_ file that configures some of the available options:
+
+```json
+{
+  "Aspire": {
+    "Azure": {
```

Comment:
The Aspire.Azure.Npgsql library is an extension to the Aspire.Npgsql library. It isn't its own standalone client integration. You configure it in the same location you configure Npgsql - `Aspire:Npgsql`.

Reply:
Fixed